controllers/storageautoscaler/scraper.go

```diff
@@ -107,16 +117,30 @@ func filterObjectsForScaling(storageAutoScalerList *ocsv1.StorageAutoScalerList,
 			err := fmt.Errorf("osd usage is nil for device class, device class: %s", storageAutoScaler.Spec.DeviceClass)
 			log.Error(err, "device class not found in sync map")
 		} else {
-			deviceClassUsage := usage.(float64)
-			if (deviceClassUsage * 100) >= float64(deviceClassThreshold) {
-				objectList = append(objectList, types.NamespacedName{Namespace: storageAutoScaler.Namespace, Name: storageAutoScaler.Name})
+			deviceClassUsage := usage.(OsdUsage)
+			if (deviceClassUsage.HighestUsage * 100) >= float64(deviceClassThreshold) {
+				if deviceClassUsage.TotalOsd == 0 {
```

Comment:
It's best to check for 0 in the same place of the potential divide by zero method. So I'd suggest to move this check into the `totalUsageGreaterThanAdjustedThreshold()` method, log the message, and then just return false in that case. Then you can unit test that case too.

Reply:
we have a another log message at line 132, `highest osd usage is greater than the threshold, but t` 

where we again need to divide it by osd_count, so thats why had a big if with it,

Not sure how can we move that log, but will think again

Reply:
How about add a helper method for `averageOsdUsage()`, and then it can check for the divide by 0. If number of OSDs is 0, it could just log the error and return 0. That helper can be called from both the log message and `totalUsageGreaterThanAdjustedThreshold()`